packages/celer-web-app/src/ui/components/Settings/SettingsConfig.tsx

```diff
@@ -0,0 +1,184 @@
+import produce from "immer";
+import React, { FunctionComponentElement } from "react";
+import { SplitType } from "core/compiler";
+import { Setting } from "core/context";
+import { MapDisplayModes, MapValues, Themes, ThemeValues } from "core/settings";
+import { SettingLabel, SettingToggle, SettingProps, SettingDropdown } from "./SettingsMenuItems";
+
+export const StringToSetting: {[key: string]: React.FC<SettingProps>} = {
+	settinglabel: SettingLabel,
+	settingtoggle: SettingToggle,
+	settingdropdown: SettingDropdown
+};
+
+export type ConfigSetting = {
+	component: string,
```

Comment:
Instead of `string`, we could use literal type (`type ConfigComponent = "settinglabel" | "settingtoggle" ....`) or an enum type for compile time validation